specification.md

```diff
@@ -344,3 +337,183 @@ In this case, if the `source` value is `source-abc` and the `booking-referral` i
+
+```js
+{
+  "resourceType": "Location",
+  "id": "456",
+  "meta": {
+    "source": "https://api.flynnspharmacy.org/fhir/smart-scheduling/Location/123"
+  },
+  "identifier": [
+    {
+      "system": "https://cdc.gov/vaccines/programs/vtrcks",
+      "value": "CV1654321"
+    },
+    {
+      "system": "https://flynnspharmacy.org/",
+      "value": "123"
+    }
+  ],
+  "name": "Flynn's Pharmacy in Pittsfield, MA",
+  // additional Location fields here...
+}
+```
+
+
+### Data “Freshness”
+
+Aggregators may request or receive information from publishers at different times, and understanding how recently data about a Location, Schedule, or Slot was retrieved can help end users gauge the accuracy of items in an aggregated dataset. _Slot Aggregators_ SHOULD use the `lastSourceSync` extension on the `meta` field of any resource to indicate the last time at which the data was known to be accurate:
+
+```js
+{
+  "resourceType": "Schedule",
+  "id": "123",
+  "meta": {
+    "extension": [{
+      "url": "http://hl7.org/fhir/StructureDefinition/lastSourceSync",
+      "valueDateTime": "2021-04-19T20:35:05.000Z"
+    }]
+  }
+  // additional Schedule fields here...
+}
+```
+
+**Note:** `lastSourceSync` has been tentatively approved, but is not yet finalized as part of FHIR as of April 28, 2021. (You can keep up-to-date on the status of this extension in FHIR’s issue tracker at [FHIR-31567][].)
+
+
+### Unknown Availability, Capacity, or Slot Times
+
+Because source systems may experience errors or may not conform to the SMART Scheduling Links specification, _Slot Aggregators_ need additional tools to describe unusual situations that are not relevant to first-part _Slot Publishers_. Specifically, _Slot Aggregators_ may describe locations with:
+- Unknown slot availability (e.g. because a source system is unreachable),
+- Unknown capacity or times (e.g. slots are known to be free or busy only _at some time_ in the near future).
+
+Features and guidelines for these scenarios SHALL NOT be used by first-party _Slot Publishers_; they are described only for use by _Slot Aggregators_.
+
+
+#### Unknown Availability
+
+If there is no source of appointment data for a Schedule or a source system is unreachable, has errors, or cannot be reliably handled for some reason, a _Slot Aggregator_ should publish a single slot for the Schedule with a `status` value of `"free"` and a “Capacity” extension with a value of `0`. First-party _Slot Publishers_ SHOULD NOT publish a slot with these features; if a slot has no free capacity, the slot should have a `status` value of `"busy"`.
+
+Example usage:
+
+```json
+{
+  "resourceType": "Slot",
+  "id": "789",
+  "schedule": {
+    "reference": "Schedule/456"
+  },
+  "status": "free",
+  "start": "2021-03-10T00:00:00Z",
+  "end": "2021-03-24T00:00:00Z",
+  "extension": [{
+    "url": "http://fhir-registry.smarthealthit.org/StructureDefinition/slot-capacity",
+    "valueInteger": 0
+  }]
+}
+```
```

Comment:
This is what we’re currently doing at USDR, but this solution feels really awkward to me. I wonder if this would be better handled by using the “has availability” extension on the Schedule with no `valueBoolean` or by changing it to use `valueCode` instead (with codes like `"yes" | "no" | "unknown"`).

Reply:
Ah, yes -- I like this. Might say `some`, `none` and `unknown` rather than "yes"/"no".